src/pages/picture-of-the-day/picture-of-the-day.test.js

```diff
@@ -0,0 +1,150 @@
+  });
+
+  it("Date picker has a calendar grid", () => {
+    setup();
+
+    const datePicker = screen.queryByRole("grid");
+
+    expect(datePicker).not.toBeInTheDocument();
+
+    fireEvent.click(
+      screen.getByRole("textbox", {
+        name: /choose date, selected date is/i,
+      })
+    );
+
+    const datePickerAfter = screen.queryByRole("grid");
+
+    expect(datePickerAfter).toBeInTheDocument();
+  });
+
+  it("User can change date", () => {
+    setup();
+
+    fireEvent.click(
+      screen.getByRole("textbox", {
+        name: /choose date, selected date is/i,
+      })
+    );
+
+    const datePicker = screen.getByRole("button", {
+      name: /calendar view is open, go to text input view/i,
+    });
+
+    fireEvent.change(datePicker, { target: { value: "2022/04/25" } });
+
+    expect(datePicker.getAttribute("value")).toBe("2022/04/25");
+  });
+});
+
+describe("Show different values reactions", () => {
+  const setState = jest.fn();
+  const useStateDateMock = () => [new Date("1995/06/01"), setState];
+  const useStateUndefinedDateMock = () => [undefined, setState];
+  const useStatePictureMock = () => [new Date([]), setState];
+  const useStateLoadingMock = () => [false, setState];
+  const useStateErrorsMock = () => [
+    "Date must be between Jun 16, 1995 and May 01, 2022.",
+    setState,
+  ];
+  const useStateNoErrorsMock = () => [false, setState];
+
+  it("User enters out Of range date", () => {
+    jest
+      .spyOn(React, "useState")
+      .mockImplementationOnce(useStateDateMock)
+      .mockImplementationOnce(useStatePictureMock)
+      .mockImplementationOnce(useStateLoadingMock)
+      .mockImplementationOnce(useStateErrorsMock);
+
+    setup();
+
+    const errorMsg = screen.getByText(
+      /date must be between jun 16, 1995 and may 01, 2022./i
+    );
+
+    expect(errorMsg).toBeInTheDocument();
+  });
+
+  it("User remove DatePicker value", () => {
+    jest
+      .spyOn(React, "useState")
+      .mockImplementationOnce(useStateUndefinedDateMock)
+      .mockImplementationOnce(useStatePictureMock)
+      .mockImplementationOnce(useStateLoadingMock)
+      .mockImplementationOnce(useStateNoErrorsMock);
+
+    setup();
+
+    fireEvent.click(
+      screen.getByRole("textbox", {
+        name: /choose date/i,
+      })
+    );
+
+    const datePicker = screen.getByRole("button", {
+      name: /calendar view is open, go to text input view/i,
+    });
+    expect(datePicker.getAttribute("value")).toBe(null);
+  });
+
+  /* it('Render hook', () => {
+
+    jest
+      .spyOn(React, "useState")
+      .mockImplementationOnce(useStateUndefinedDateMock)
+      .mockImplementationOnce(useStatePictureMock)
+      .mockImplementationOnce(useStateLoadingMock)
+      .mockImplementationOnce(useStateNoErrorsMock);
+
+
+    const { result } = renderHook(() => useGetData())
+
+    act(() => {
+      result.current.fetchData();
+    })
+
+    expect(result.current.date).toBe('')
+  }) */

```

Comment:
This looks like you tried to test the hook

Testing hook can be tricky, it not intuitive. The coverage of simple hooks can be 100% without the need to write an specific test for the hook, but some hooks will lower the coverage if there is not a test for specific cases of it